src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -200,6 +200,9 @@ internal static string DecompressToken(byte[] tokenBytes, string algorithm)
                 {
                     Validators.ValidateAlgorithm(decryptionParameters.Enc, key, jwtToken, validationParameters);
                     decryptedTokenBytes = DecryptToken(cryptoProviderFactory, key, decryptionParameters);
+                    X509SecurityKey x509Key = key as X509SecurityKey;
+                    if (x509Key != null)
+                        LogHelper.LogInformation(TokenLogMessages.IDX10903, x509Key.Certificate?.Thumbprint);
```

Comment:
Wondering if there is anything from the decryptionParameters we can log which will give similar information? or if we should log that regardless, just in case this is not an x509 cert/key. or perhaps from the cryptoprovider?

Reply:
also considering if we should put all of it behind an Log.IsInformation check (right now it's just a typecast and a comparison, but it will be waste of cpu cycles in most cases as it's not recommended to run with informational logging (I believe).

Reply:
x509Key.Certificate cannot be null.